### ISSUE TITLE
A question about trasplanting the model to KITTI dataset.

### ISSUE THREAD

**Issue by songjiaxina** [mikacuy/pointnetvlad#3]
Hi! Thanks for your great work and code release. Recently I am also doing some research on 3d-relocalization.  After reading your paper and part of code, I am trying to convert the kitti data to the format which this code can run.  And I wonder how strong is the generalization ability of this model? Will the recall decrease much? Have you tested before? Great thanks!!

**Comment by mikacuy** [comment]
Hi. I haven't tested on KITTI before because the work mainly uses datasets where an area is repeatedly visited multiple times under different weather/lighting conditions. That's why the Oxford dataset was chosen as the initial benchmark. 

When training on Oxford and then testing on the inhouse datasets, there was a decrease in performance of about ~15%, but top 1% accuracy was still above 60%, so the method should be feasible. But after fine-tuning, the results went back up again. I suspect that issue is not only on generalization of the model design, but also that the training data is limited (especially compared to image datasets) that caused the decrease in performance when tested on unseen datasets.

**Comment by songjiaxina** [comment]
OK! Great! Thanks for your details!!

**Comment by kxhit** [comment]
@songjiaxina Hi! Have you tested the code on KITTI dataset? Is there any preprocessing code of KITTI dataset?
Thanks!

**Comment by kxhit** [comment]
Hi! @mikacuy @songjiaxina 
I test the refined pretrained model on KITTI odometry dataset. I get the submaps by using ground truth poses and preprocessing similar to the paper. The PR curve appears good. However, PointNetVLAD seems not robust enough to rotation. Note that sequence 08 has reverse loop.
I will try to train on KITTI dataset and I believe the results will be better. 

sequence 00:
![00_PV_pr_curve](https://user-images.githubusercontent.com/16812362/66574650-31dacf00-eba7-11e9-8f67-2cabcef01abe.png)

with random rotation around z-axis (yaw):
![00_PV_pr_curve](https://user-images.githubusercontent.com/16812362/66574887-9bf37400-eba7-11e9-9a92-f54fb4d5ee99.png)

sequence 02:
![02_PV_pr_curve](https://user-images.githubusercontent.com/16812362/66574943-b4fc2500-eba7-11e9-96b1-ed620c72cb06.png)

with random rotation around z-axis (yaw):
![02_PV_pr_curve](https://user-images.githubusercontent.com/16812362/66574929-af9eda80-eba7-11e9-83fa-ea32672b108d.png)

sequence 05:
![05_PV_pr_curve](https://user-images.githubusercontent.com/16812362/66574992-cb09e580-eba7-11e9-9edb-fec4c4f69c12.png)

with random rotation around z-axis (yaw):
![05_PV_pr_curve](https://user-images.githubusercontent.com/16812362/66575017-d52be400-eba7-11e9-98aa-29a2dd8783d4.png)


sequence 08:
![08_PV_pr_curve](https://user-images.githubusercontent.com/16812362/66575057-e1b03c80-eba7-11e9-9582-003c0ab01972.png)

with random rotation around z-axis (yaw):
![08_PV_pr_curve](https://user-images.githubusercontent.com/16812362/66575093-ef65c200-eba7-11e9-8594-5fcbff47a00d.png)
